spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -894,6 +895,10 @@ private Table load(Identifier ident) {
         }
       }
 
+      if (ident.name().equals(REWRITE)) {
```

Comment:
Nit: The changelog table case uses `equalsIgnoreCase`. Probably good to have identical logic for selectors.